packages/flutter/test/material/app_bar_test.dart

```diff
@@ -1027,6 +1027,29 @@ void main() {
     expect(find.byIcon(Icons.menu), findsNothing);
   });
 
+  testWidgets('AppBar sizes according to toolbarHeight', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
+    const double customToolbarHeight = 128.0;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: AppBar(
+            leading: Placeholder(key: key),
+            title: const Text('Abc'),
+            toolbarHeight: customToolbarHeight,
+            actions: const <Widget>[
+              Placeholder(fallbackWidth: 10.0),
+              Placeholder(fallbackWidth: 10.0),
+              Placeholder(fallbackWidth: 10.0),
+            ],
+          ),
+        ),
+      ),
+    );
+    expect(tester.renderObject<RenderBox>(find.byKey(key)).localToGlobal(Offset.zero), const Offset(0.0, 0.0));
```

Comment:
It looks like there are some helper functions here already for getting values from the AppBar, see `appBarHeight`, `appBarTop`, appBarBottom` in this file.